rir/src/compiler/pir/instruction.h

```diff
@@ -193,6 +193,27 @@ class Instruction : public Value {
         }
     }
 
+    // Won't produce errors if all arguments are type argt (and no env)
+    // Won't produce warnings if all arguments are argt and scalar (and no env)
+    void maskWarnErrOn(PirType argt) {
+        bool noErr = true;
+        bool noWarn = true;
+        eachArg([&](Value* v) {
+            if (mayHaveEnv() && env() == v)
+                return;
+            if (!v->type.isA(argt)) {
+                noErr = false;
+                noWarn = false;
+            } else if (!v->type.isScalar()) {
```

Comment:
ok, this is strange. this is going to bite somebody, since it is so unexpected. why is this extra case needed? This seems to be highly specific to particular instructions. wouldn't it be better to just add this to the custom `updateType` implementation of the instructions that need it?